### ISSUE TITLE
Add Triton/torch.compile perf models (parity with ATen perf models)

### ISSUE THREAD

**Issue by ajassani** [AMD-AGI/TraceLens#619]
### Background

TraceLens currently has no performance models for `torch.compile`-generated Triton kernels. They show up in the trace as opaque `triton_*` kernels with measured time but no FLOPs, bytes, AI, or achieved-perf — unlike `aten::mm` and other ops that already have perf models and populate `unified_perf_summary`.

Concrete example — `unified_perf_summary` from a small transformer block (RMSNorm + Attention + SwiGLU MLP, B=8, T=4096, dim=2048, bf16) generated through TraceLens today:

| kernel                                               | cat      | kernel_time_us | % gpu | GFLOPS | TFLOPS/s | TB/s | perf model |
|:-----------------------------------------------------|:---------|---------------:|------:|:-------|:---------|:-----|:-----------|
| aten::_scaled_dot_product_flash_attention            | SDPA_fwd |          26637 |  29.0 | 549.8  | 206.4    | 0.20 | yes        |
| aten::mm                                             | GEMM     |          25946 |  28.3 | 755.9  | 583.4    | 0.41 | yes        |
| aten::mm                                             | GEMM     |          15614 |  17.0 | 824.6  | 528.3    | 0.36 | yes        |
| aten::mm                                             | GEMM     |          12216 |  13.3 | 755.9  | 619.2    | 0.43 | yes        |
| aten::mm                                             | GEMM     |           4770 |   5.2 | 274.9  | 576.3    | 0.58 | yes        |
| triton_poi_fused_mul_silu_2                          | triton   |           3241 |   3.5 | —      | —        | —    | no         |
| triton_poi_fused_add_3                               | triton   |           1364 |   1.5 | —      | —        | —    | no         |
| triton_red_fused__to_copy_add_mean_mul_pow_rsqrt_1   | triton   |           1114 |   1.2 | —      | —        | —    | no         |
| triton_red_fused__to_copy_add_mean_mul_pow_rsqrt_0   | triton   |            908 |   1.0 | —      | —        | —    | no         |

ATen rows have full perf metrics; Triton rows are blank. Even on this minimal block ~7% of GPU time lands in kernels we can't characterize; on real workloads with more aggressive fusion the share grows.

> Read the table for *structure* — which rows have perf-model data, which don't. Specific numbers are from a small minimal repro and aren't meant as a benchmark.

#### Repro

1. Define a small transformer block (RMSNorm + attention + SwiGLU MLP works), wrap with `torch.compile(..., mode="default", fullgraph=True)`, run a forward pass on dummy input shaped `[8, 4096, 2048]` in bf16 under `torch.profiler.profile(activities=[CPU, CUDA], record_shapes=True)`, export Chrome trace.
2. Run TraceLens on the captured trace:
   ```bash
   TraceLens_generate_perf_report_pytorch \
       --profile_json_path trace.json.gz \
       --output_xlsx_path perf_report.xlsx
   ```
3. Open the `unified_perf_summary` sheet and look at the `triton_*` rows — `GFLOPS`, `TFLOPS/s`, `TB/s` are blank.

### Goal

Add a perf-model layer for `torch.compile`-generated Triton kernels so they participate in `unified_perf_summary` the same way ATen ops do:

- `triton_*` kernels populate the same columns as ATen entries: `GFLOPS`, `Data Moved (MB)`, `FLOPS/Byte`, `TFLOPS/s`, `TB/s`
- They show up alongside ATen ops in the perf-metrics view, not as a blind spot
- Numbers follow the same TraceLens philosophy as the existing perf models: **load each input once, write each output once, full reuse, no redundant FLOPs** (the "useful FLOPs / useful bytes" assumption documented in `perf_report_columns.md` §3)

### Approach (starting point — high level)

Parse the `torch.compile` debug log (the artifacts Inductor emits with `TORCH_COMPILE_DEBUG=1`) to recover what each `triton_*` kernel computes, then map that to FLOPs and bytes using the existing TraceLens perf-model conventions. Implementation details to be figured out by the implementer.

### Scope (v1, parity goal)

- General dispatch logic that handles arbitrary fusions of known ATen ops, not a hardcoded list of named patterns. The constituent ATen ops are recoverable from the kernel name / Inductor artifacts; the perf model for the fused kernel is composed from the per-op contributions plus the fused I/O.
- Use the small transformer block above (RMSNorm, residual-add+RMSNorm, SwiGLU silu*mul, residual-add, etc.) as a starting point for development and validation, but the logic should generalize — adding a new ATen op to the supported set should grow coverage without per-fusion code.
- Integrate with the existing perf-metrics pipeline so reports populate the new entries automatically.

### Long-term direction

The `torch.compile` log has lots more actionable info — fusion decisions, recompiles, graph breaks, padding/stride adjustments, autotuning configs. Perf models is the priority (unblocks roofline). Other extractions can be scoped as follow-ups as patterns emerge during implementation.

### References

- Existing ATen perf models in `TraceLens/PerfModel/perf_model.py` — same philosophy and dispatch pattern to follow

### Definition of done

- General dispatch logic in place; `triton_*` kernels in the test transformer block appear in the perf-metrics sheets with FLOPs/bytes/AI/achieved-perf populated
- Tested on the small transformer block + at least one real workload trace (e.g., vLLM forward pass)

**Comment by ajassani** [comment]
Minimal repro script for the trace shown in the issue body. Drop in a file (`profile_repro.py`), run, get a Chrome-format trace ready for `TraceLens_generate_perf_report_pytorch`.

```python
"""Tiny torch.compile transformer block under torch.profiler.

Captures a Chrome trace with kernel-level timing for the inductor-generated
triton_* kernels (RMSNorm, residual-add, SwiGLU silu*mul, etc.).

Run:
    python profile_repro.py
Output:
    ./profile_out/trace.json.gz
"""
import os
import torch
import torch.nn as nn
import torch.nn.functional as F
from torch.profiler import profile, ProfilerActivity, schedule


class RMSNorm(nn.Module):
    def __init__(self, dim, eps=1e-6):
        super().__init__()
        self.weight = nn.Parameter(torch.ones(dim))
        self.eps = eps

    def forward(self, x):
        var = x.float().pow(2).mean(-1, keepdim=True)
        x = x * torch.rsqrt(var + self.eps).to(x.dtype)
        return x * self.weight


class SwiGLUMLP(nn.Module):
    def __init__(self, dim, hidden):
        super().__init__()
        self.gate = nn.Linear(dim, hidden, bias=False)
        self.up = nn.Linear(dim, hidden, bias=False)
        self.down = nn.Linear(hidden, dim, bias=False)

    def forward(self, x):
        return self.down(F.silu(self.gate(x)) * self.up(x))


class Attention(nn.Module):
    def __init__(self, dim, heads):
        super().__init__()
        self.heads = heads
        self.head_dim = dim // heads
        self.qkv = nn.Linear(dim, 3 * dim, bias=False)
        self.out = nn.Linear(dim, dim, bias=False)

    def forward(self, x):
        B, T, D = x.shape
        qkv = self.qkv(x).view(B, T, 3, self.heads, self.head_dim)
        q, k, v = qkv.unbind(dim=2)
        q, k, v = (t.transpose(1, 2) for t in (q, k, v))
        y = F.scaled_dot_product_attention(q, k, v, is_causal=True)
        y = y.transpose(1, 2).contiguous().view(B, T, D)
        return self.out(y)


class Block(nn.Module):
    def __init__(self, dim, heads, hidden):
        super().__init__()
        self.norm1 = RMSNorm(dim)
        self.attn = Attention(dim, heads)
        self.norm2 = RMSNorm(dim)
        self.mlp = SwiGLUMLP(dim, hidden)

    def forward(self, x):
        x = x + self.attn(self.norm1(x))
        x = x + self.mlp(self.norm2(x))
        return x


def main():
    B, T, dim, heads, hidden = 8, 4096, 2048, 16, 5632
    dtype = torch.bfloat16
    device = "cuda"

    torch.manual_seed(0)
    block = Block(dim, heads, hidden).to(device=device, dtype=dtype)
    x = torch.randn(B, T, dim, device=device, dtype=dtype)
    compiled = torch.compile(block, mode="default", fullgraph=True)

    with torch.inference_mode():
        for _ in range(5):
            _ = compiled(x)
        torch.cuda.synchronize()

    out_dir = os.path.join(os.getcwd(), "profile_out")
    os.makedirs(out_dir, exist_ok=True)
    trace_path = os.path.join(out_dir, "trace.json.gz")

    sched = schedule(wait=0, warmup=1, active=10, repeat=1)
    with profile(
        activities=[ProfilerActivity.CPU, ProfilerActivity.CUDA],
        schedule=sched,
        record_shapes=True,
    ) as prof:
        with torch.inference_mode():
            for _ in range(11):
                _ = compiled(x)
                prof.step()
        torch.cuda.synchronize()
    prof.export_chrome_trace(trace_path)
    print(f"trace: {trace_path}")


if __name__ == "__main__":
    main()
```

Then run TraceLens on the trace:

```bash
TraceLens_generate_perf_report_pytorch \
    --profile_json_path profile_out/trace.json.gz \
    --output_xlsx_path perf_report.xlsx
```

Open `unified_perf_summary` and the `triton_*` rows are the ones to fill in.

#### Dumping the torch.compile debug log

To get the Inductor debug artifacts (`output_code.py`, `ir_post_fusion.txt`, `fx_graph_*.py`), re-run the same script with debug output enabled:

```bash
TORCH_COMPILE_DEBUG=1 \
TORCH_COMPILE_DEBUG_DIR=./tcd \
python profile_repro.py
```

Files land under `./tcd/run_*-pid_*/torchinductor/<model>__*/`.
